jgrapht-core/src/main/java/org/jgrapht/generate/RandomRegularGraphGenerator.java

```diff
@@ -0,0 +1,236 @@
+ * runs in expected $\mathcal{O}(nd^2)$ time.
+ * It has been proved in [KV03] to sample from the space of random d-regular graphs
+ * in a way which is asymptotically uniform at random when $d = \mathcal{O}(n^{1/3 - \epsilon})$.
+ *
+ * <p>
+ * [KV03] Kim, Jeong Han, and Van H. Vu.
+ * "Generating random regular graphs."
+ * Proceedings of the thirty-fifth annual ACM symposium on Theory of computing. ACM, 2003.
+ *
+ * [SW99] Steger, Angelika, and Nicholas C. Wormald.
+ * "Generating random regular graphs quickly."
+ * Combinatorics, Probability and Computing 8.4 (1999): 377-396.
+ *
+ * @author Emilio Cruciani
+ * @since March 2018
+ *
+ * @param <V> graph node type
+ * @param <E> graph edge type
+ */
+public class RandomRegularGraphGenerator<V, E> implements GraphGenerator<V, E, V> {
+
+    private final int n;
+    private final int d;
+
+    /**
+     * Construct a new RandomRegularGraphGenerator.
+     *
+     * @param n number of nodes
+     * @param d degree of nodes
+     * @throws IllegalArgumentException if number of nodes is negative
+     * @throws IllegalArgumentException if degree is negative
+     * @throws IllegalArgumentException if degree is greater than number of nodes
+     * @throws IllegalArgumentException if the value "n * k" is odd
+     */
+    public RandomRegularGraphGenerator(int n, int d) {
+        if (n < 0) {
+            throw new IllegalArgumentException("number of nodes must be non-negative");
+        }
+        else if (d < 0) {
+            throw new IllegalArgumentException("degree of nodes must be non-negative");
+        }
+        else if (d > n) {
+            throw new IllegalArgumentException("degree of nodes must be smaller than or equal to number of nodes");
+        }
+        else if ((n * d) % 2 != 0) {
+            throw new IllegalArgumentException("value 'n * k' must be even");
+        }
+        else {
+            this.n = n;
+            this.d = d;
+        }
+    }
+
+    /**
+     * Generate a random regular graph.
+     *
+     * @param target target graph
+     * @param vertexFactory vertex factory
+     * @param resultMap result map
+     * @throws IllegalArgumentException if target is directed
+     * @throws IllegalArgumentException if "d == n" and graph is simple
+     */
+    @Override
+    public void generateGraph(Graph<V, E> target, VertexFactory<V> vertexFactory, Map<String, V> resultMap) {
+
+        // directed case
+        if (target.getType().isDirected()) {
+            throw new IllegalArgumentException("target graph must be undirected");
+        }
+
+        // simple case
+        else if (target.getType().isSimple()) {
+            // no nodes or zero degree case
+            if (this.n == 0 || this.d == 0) {
+                EmptyGraphGenerator<V, E> emptyGraphGenerator = new EmptyGraphGenerator<>(this.n);
+                emptyGraphGenerator.generateGraph(target, vertexFactory, resultMap);
+            }
+
```

Comment:
See file `etc/eclipse-formatter-settings.xml` for the coding conventions used in the library. We automatically enforce them on each release, but still, following them is nice. In case you use Eclipse for development, you can load the formatter settings so that they are applied automatically.

Reply:
I am actually using IntelliJ IDEA, but I imported the formatter-settings.
Regarding the code above, what is unconventional? I am not getting it.